postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresSchemaReader.java

```diff
@@ -43,6 +46,14 @@ public Schema getSchema(ResultSetMetaData metadata, int index) throws SQLExcepti
     String typeName = metadata.getColumnTypeName(index);
     int columnType = metadata.getColumnType(index);
 
+    // raise exception if scale is not set for decimal type
+    if (columnType == 2) {
+      if (metadata.getScale(index) == 0) {
+        throw new ValidationException(Collections.singletonList(
```

Comment:
This method is being called from ``AbstractDBSource->getSchema->loadSchemaFromDB->getSchemaReader().getSchemaFields`` which is located in ``CommonSchemaReader`` and calls ``getSchema``.

We could not find a way to access failure collector from here.

Reply:
This logic should be added in `CommonSchemaReader#getSchema()`. When scale is 0 then throw checked exception and handle it in caller `AbstractDBSource`

Reply:
As per our offline discussion we need to handle following cases:

1.) Numeric with precision and scale
If scale is zero we should throw a validation error as we dont support scale to be zero

2.) Numeric with precision
By default, it will have scale zero. So we should throw a validation error.

3.) Numeric with varying precision and scale in a given column
This case we will not support. So throw an error if columns have varying scale and precision.

How are we handling #3?

Reply:
updated PR. As for handling #3 - it was already there as ``setDecimal`` method in ``StructuredRecord`` raises the following error :
![375-PostreSQL-Decimal-scale-mistmatch](https://user-images.githubusercontent.com/7079754/97717251-e0a8c080-1ac4-11eb-8a56-15b6095db965.png)

Reply:
When reading column metadata during schema detection for columns with dynamic scale and precision both are returned as 0 which is caught by current validation condition for scale 0. 
However if user sets output schema manually and sets custom scale and precision value then the dynamic scale issue can only be detected while trying to match the actual record schema to output schema (which is happens currently).
I don't see how we can catch this without reading a record that has mismatch with output schema.

Reply:
I see. In that case can you add comment in the code explaining behavior in all 3 cases?